src/map/mail.cpp

```diff
@@ -99,13 +100,14 @@ int mail_removeitem(map_session_data *sd, short flag, int idx, int amount)
 	return 1;
 }
 
-bool mail_removezeny( map_session_data *sd, bool flag ){
+bool mail_removezeny(map_session_data *sd, bool flag) {
 	nullpo_retr( false, sd );
 
 	if( sd->mail.zeny > 0 ){
 		//Zeny send
 		if( flag ){
-			if( pc_payzeny( sd, sd->mail.zeny + sd->mail.zeny * battle_config.mail_zeny_fee / 100, LOG_TYPE_MAIL, NULL ) ){
+			// It's possible that we don't know what the target_id is, so it will be 0
+			if (pc_payzeny(sd, sd->mail.zeny + sd->mail.zeny * battle_config.mail_zeny_fee / 100, LOG_TYPE_MAIL, sd->mail.target_id)) {
```

Comment:
We don't do this for vendor tax, so I just left it as is

Reply:
I just brought it up because of what @Tokeiburu wrote in the issue:
`Currently, you have to divide the sender's amount of zeny by the RODex tax, then you have to look through the logs searching for that specific value with a date close to the sender's.`

Reply:
We won't have to worry about that, as we log the sender losing total+tax to dest_id, and the receiver gaining total from send_id